yt_dlp/YoutubeDL.py

```diff
@@ -2586,6 +2586,9 @@ def _fill_common_fields(self, info_dict, final=True):
                     upload_date = datetime.datetime.fromtimestamp(info_dict[ts_key], datetime.timezone.utc)
                     info_dict[date_key] = upload_date.strftime('%Y%m%d')
 
+        if info_dict.get('release_year') is None and info_dict.get('release_date') is not None:
+            info_dict['release_year'] = int_or_none(info_dict['release_date'][:4])
```

Comment:
The thing that gives me pause about this is that it will make so many otherwise-passing download tests now fail

Also could probably just use `int()` here, but either way is fine

Reply:
We can make an exception in the tester similar to https://github.com/yt-dlp/yt-dlp/blob/21dc069bea2d4d99345dd969e098f4535c751d45/test/helper.py#L221-L223

Reply:
```suggestion
        if not info_dict.get('release_year'):
            info_dict['release_year'] = traverse_obj(info_dict, ('release_date', lambda x: int(x[:4])))
```

Reply:
Okay, so should this be used as an exception in `sanitize_got_info_dict`?
```python
    # release_year may be generated from release_date
    if test_info_dict.get('release_year') and test_info_dict.get('release_date'):
        test_info_dict.pop('release_year')
```
There shouldn't be that many IE's that return both a `release_date` and `release_year` being returned.
I could find: `ArchiveOrgIE`, `MonstercatIE`, `TrueIDIE` (inconstant), and `YoutubeIE`

Reply:
There can be situations in theory where the year doesn't match the date. I think this'd be better:

```py
    if try_call(lambda: test_info_dict['release_year'] == int(test_info_dict['release_date'][:4])):
        test_info_dict.pop('release_year')
```

Reply:
> and `YoutubeIE`

release_year is currently broken for Youtube if I remember right